src/components/ride/rideService.js

```diff
@@ -0,0 +1,50 @@
+import { logger } from '../../middleware/logger.js';
+import { catchAsync } from '../../utils/catchAsync.js';
+import ExpressError from '../../utils/ExpressError.js';
+
+export const createNewRide = (Model) =>
+  catchAsync(async (req, res) => {
+    const { ride, driverName, location } = req.body;
+    const newRide = await Model.create({ ride, driverName, location });
+
+    res.json({
+      status: 'Success',
+      data: {
+        newRide,
+      },
+    });
+  });
+
+export const getRides = (Model) =>
+  catchAsync(async (req, res) => {
+    const { distance, latlng } = req.params;
+    const [lat, lng] = latlng.split(','); //19.117659, 72.863165
+
+    // Calulating radius based on 'KM'
+    const radius = distance / 6378.1;
+
+    if (!lat || !lng) {
+      throw new Error(
+        'Please provide latitude and longitude in the format lat, lng'
+      );
+    }
+
+    logger.info(distance, lat, lng);
+
+    const rides = await Model.find({
```

Comment:
Use appropriate naming convention for variables. Like for user model it would be User.find({})